### ISSUE TITLE
Constructor - 1 will keep the error code in the BlockNum field -> caller must check if  the blockNum is error or not

### ISSUE THREAD

**Issue by gokulsreekumar** [Nitcbase/nitcbase.github.io#40]

**Comment by gokulsreekumar** [comment]
* Add the comment in the function to warn the Caller function to check if an error is returned or not in the `blockNum` field and if so return the appropriate error code back from the caller
* Make changes to ALL the caller's of this function to do the error check

**Comment by gokulsreekumar** [comment]
The callers of [this constructor](https://nitcbase.github.io/docs/Design/Buffer%20Layer#blockbuffer--blockbuffer-constructor1) and the following constructors: [RecBuffer :: RecBuffer() (Constructor 1)](#recbuffer--recbuffer-constructor-1), [IndBuffer :: IndBuffer() (Constructor 1)](#indbuffer--indbuffer-constructor-1), [IndInternal :: IndInternal() (Constructor1)](#indinternal--indinternal-constructor1) and [IndLeaf :: IndLeaf() (Constructor 1)](#indleaf--indleaf-constructor-1) should check the value of `blockNum` field to verify if the disk block was allocatted succesfully.

**Comment by gokulsreekumar** [comment]
**Done**,
Except for the Indexing part changes.
ba_insert() method is the only place where constructor 1 of RecBuffer is being called to allocate a new block.

Basic Idea → It is only called when we need more space
- Create Relation → This will boil down to ba_insert calls
- Create Index → Not sure about this as of now
- Insert (and Importing) Records
    - → This directly calls ba_insert
- Selecting
    - Basically this will be create relation followed by insert records
        - Thus boiling down to ba_insert call
